### DISCUSSION TITLE
SSR for SEO

### DISCUSSION THREAD

**Discussion by bard-hobson** [react-page/react-page#1228] [Q&A]
Currently the tool renders on the client side. When saving to the database it is a json object. 

This causes crawl engines such as ahrefs to think there is no content. I tried to SSR the editor json but as an object it doesn't work. Has anyone had any luck rendering the editor server side so the "post" content is viewable by a crawler?

**Comment by macrozone** [comment]
@bard-hobson SEO works out of the box. ReactPage also does not use or load any editor code on server rendering and initial page load. Editor code is only loaded on the client and only if `readOnly={false}` (or not set)

It probably has to do how you load the data for the editor and whether this works on the server. E.g use nextjs and getServerSideProps or static props. 

Our example app https://react-page.github.io/ is using nextjs and does full static site generation https://github.com/react-page/react-page/tree/master/examples

you see that the full content is there on the initial document
